leaderelection/config.go

```diff
@@ -127,25 +129,37 @@ type ComponentConfig struct {
 	RetryPeriod   time.Duration
 }
 
-// StatefulSetConfig represents the required information for a StatefulSet service.
-type StatefulSetConfig struct {
-	StatefulSetName string
-	ServiceName     string `envconfig:"STATEFUL_SERVICE_NAME" required:"true"`
-	Port            string `envconfig:"STATEFUL_SERVICE_PORT" default:"80"`
-	Protocol        string `envconfig:"STATEFUL_SERVICE_PROTOCOL" default:"http"`
+// statefulSetID is a envconfig Decodable controller ordinal and name.
+type statefulSetID struct {
+	ssName  string
+	ordinal int
 }
 
-func NewStatefulSetConfig() (*StatefulSetConfig, error) {
-	var ssc StatefulSetConfig
-	ssn, _, err := ParseControllerOrdinal()
-	if err != nil {
-		return nil, err
+func (ssID *statefulSetID) Decode(v string) error {
+	if i := strings.LastIndex(v, "-"); i != -1 {
+		ui, err := strconv.ParseUint(v[i+1:], 10, 64)
+		ssID.ordinal = int(ui)
+		ssID.ssName = v[:i]
+		return err
 	}
-	ssc.StatefulSetName = ssn
-	if err := envconfig.Process("", &ssc); err != nil {
+	return fmt.Errorf("%q is not a valid stateful set controller ordinal", v)
+}
+
+// statefulSetConfig represents the required information for a StatefulSet service.
+type statefulSetConfig struct {
+	StatefulSetID statefulSetID `envconfig:"CONTROLLER_ORDINAL" required:"true"`
```

Comment:
I wonder if we should make this `STATEFUL_ORDINAL` for consistency? 🤔

Reply:
Yeah, it was a sore in my eye a well. let's do it. Went for `STATEFUL_CONTROLLER_ORDINAL`.